plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java

```diff
@@ -39,7 +50,27 @@ public static SnykExtendedLanguageClient getInstance() {
   }
 
   public void triggerScan() {
-    ExecuteCommandParams params = new ExecuteCommandParams("snyk.workspace.scan", new ArrayList<>());
+    // run the wizard if no token
+    if (Preferences.getInstance().getAuthToken().isBlank()) {
+      SnykWizard wizard = new SnykWizard();
+
+      WizardDialog dialog = new WizardDialog(PlatformUI.getWorkbench().getDisplay().getActiveShell(), wizard);
+
+      dialog.setBlockOnOpen(true);
+      dialog.open();
+    } else {      
```

Comment:
Is it really an else? I think scan should be run after the wizard has finished, as that's the user intention.

Reply:
It can be done without the else, the problem is that we will attempt a scan after clicking `finish` in the Wizard, however `finish` will initiate the authentication user flow. This results in a scan being attempted before the authentication flow is complete.

The alternative is to prevent the scan with the `else` or to complete the auth flow before the wizard finishes (which we intentionally changed to auth on clicking finish).

Reply:
Hm... we could trigger a scan after authentication, reacting to `$/snyk.hasAuthenticated`. That would be the event-driven approach.